### PR TITLE
ref(migration): Make repositoryprojectpathconfig organization_integration foreignkey nullable

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
### Objective:

We need to create a  migration to make `organization_integration_id` nullable. This allows Code Path Mappings to be an integration-agnostic feature. This will allow orgs that do not have an integration to setup Code Path Mappings.

Code Path Mappings will become the foundation for a new CODEOWNERS file import & sync feature. And this new feature will need to work for orgs without integrations.

### CODE REVIEW

**Comment by scefali** [comment]
@NisanthanNanthakumar Making this work without an integration sounds nice but I'm curious what the backstory is here

**Comment by scefali** [comment]
For those wondering, there are only about 4k rows in the `sentry_integrationexternalproject` table for SaaS Sentry.

**Comment by MeredithAnya** [comment]
@NisanthanNanthakumar This probably doesn't need to be in this PR but currently the code mapping endpoints are nested under the organization integration:

```
organization_integration_repository_project_path_configs.py
organization_integration_repository_project_path_config_details.py
```

so we have plans to make new endpoints for this? or are we going to be including the code mapping creation only when codeowners are uploaded?

@scefali there are large users that want to be able to upload their codeowner files to us but we still need mappings to be able to covert their codeowners file to our syntax and with the proper paths. these users just won't be able to use stack trace linking

**Review by scefali — APPROVED** [review]
LGTM

File: src/sentry/models/integration.py [hunk 1]
```diff
@@ -51,7 +51,9 @@ class RepositoryProjectPathConfig(DefaultFieldsModel):
 
     repository = FlexibleForeignKey("sentry.Repository")
     project = FlexibleForeignKey("sentry.Project", db_constraint=False)
-    organization_integration = FlexibleForeignKey("sentry.OrganizationIntegration")
+    organization_integration = FlexibleForeignKey(
+        "sentry.OrganizationIntegration", null=True, on_delete=models.SET_NULL
```
Comment (wedamija) [inline src/sentry/models/integration.py:55]: Just checking that `SET_NULL` is really what you want here? So you want a `RepositoryProjectPathConfig` row to be orphaned from that relationship when a related `OrganizationIntegration` is deleted, is that correct?

**Review by wedamija — COMMENTED** [review]
Migration looks fine to me. I checked and this doesn't look like it's used as part of ingest, so should be safe to run.

File: src/sentry/models/integration.py (hunk 1 above)
Comment (NisanthanNanthakumar) [reply src/sentry/models/integration.py:55]: @wedamija ahh yes that's what I wanted, but I just spoke with @MeredithAnya about it and agreed that its better to delete the `RepositoryProjectPathConfig` row. I'll make an update for it.

**Comment by NisanthanNanthakumar** [comment]
```
BEGIN;
--
-- Alter field organization_integration on repositoryprojectpathconfig
--
SET CONSTRAINTS "sentry_repositorypro_organization_integra_d70daa81_fk_sentry_or" IMMEDIATE; ALTER TABLE "sentry_repositoryprojectpathconfig" DROP CONSTRAINT "sentry_repositorypro_organization_integra_d70daa81_fk_sentry_or";
ALTER TABLE "sentry_repositoryprojectpathconfig" ALTER COLUMN "organization_integration_id" DROP NOT NULL;
ALTER TABLE "sentry_repositoryprojectpathconfig" ADD CONSTRAINT "sentry_repositorypro_organization_integra_d70daa81_fk_sentry_or" FOREIGN KEY ("organization_integration_id") REFERENCES "sentry_organizationintegration" ("id") DEFERRABLE INITIALLY DEFERRED;
COMMIT;
```

**Review by wedamija — APPROVED** [review]

**Pull request merged**